hazelcast/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java

```diff
@@ -126,6 +142,76 @@ public Job newJobProxy(long jobId, boolean isLightJob, @Nonnull Object jobDefini
         return new ClientJobProxy(client, jobId, isLightJob, jobDefinition, config);
     }
 
+    @Override
+    public void submitJobFromJar(@Nonnull Path jarPath, String snapshotName, String jobName, String mainClass,
+                                 @Nonnull List<String> jobParameters) {
+
+        try {
+            UUID sessionId = UuidUtil.newSecureUUID();
+
+            String fileName = fileNameWithoutExtension(jarPath);
+            String sha256Hex = calculateSha256Hex(jarPath);
+            long jarSize = Files.size(jarPath);
+
+            // Send job meta data
+            boolean result = sendJobMetaData(sessionId, fileName, sha256Hex, snapshotName, jobName, mainClass,
+                    jobParameters);
+            if (result) {
+                logFine(getLogger(), "Submitted JobMetaData successfully for jarPath: %s", jarPath);
+            }
+            // Send job parts
+            sendJobMultipart(jarPath, sessionId, jarSize);
```

Comment:
What's going to happen when we lose connection for a while and server clears the already uploaded parts. Do we get some meaningful exception?

Reply:
`if (jobUploadStatus == null) {
   throw new JetException("Unknown session id : " + sessionId);
 }`